vowpalwabbit/slim/include/vw_slim_predict.h

```diff
@@ -372,7 +484,118 @@ class vw_predict
     uint64_t weight_length = (uint64_t)1 << _num_bits;
     _stride_shift = (uint32_t)ceil_log_2(num_weights);
 
-    RETURN_ON_FAIL(mp.read_weights<W>(_weights, _num_bits, _stride_shift));
+    RETURN_ON_FAIL(mp.read_weights<W>(_predict_info->_weights, _num_bits, _stride_shift));
+
+    // TODO return error if activated but these are not initialized
+    if (pmf_to_pdf != nullptr && cats_tree != nullptr)
+    {
+      // TODO plug in interactions reduction and maybe metrics (or maybe figure out how to use setup functions)
+
+      // cats_tree -> binary -> gd
+      // create gd
+      auto* ret = VW::LEARNER::make_base_learner(
```

Comment:
you might be able to leverage setup_base_i, maybe with custom setup functions specialized for slim? Just a thought, not sure if its actually useful here or not

Reply:
Yes that might be the way to go

Reply:
yes, you mentioned this a while back. I didn't realize it was about this PR